Userland/Libraries/LibWeb/HTML/BrowsingContext.h

```diff
@@ -175,47 +146,45 @@ class BrowsingContext final
 
-    virtual String const& window_handle() const override { return m_window_handle; }
-    virtual void set_window_handle(String handle) override { m_window_handle = move(handle); }
+    void set_is_popup(TokenizedFeature::Popup is_popup) { m_is_popup = is_popup; }
 
 private:
-    explicit BrowsingContext(JS::NonnullGCPtr<Page>, HTML::NavigableContainer*);
+    explicit BrowsingContext(JS::NonnullGCPtr<Page>);
 
     virtual void visit_edges(Cell::Visitor&) override;
 
     void reset_cursor_blink_cycle();
 
     JS::NonnullGCPtr<Page> m_page;
 
+    // FIXME: Move EventHandler to Navigable
     Web::EventHandler m_event_handler;
 
-    // https://html.spec.whatwg.org/multipage/history.html#current-entry
-    SessionHistoryEntry& current_entry() { return m_session_history[*m_session_history_index]; }
-    SessionHistoryEntry const& current_entry() const { return m_session_history[*m_session_history_index]; }
-    Optional<size_t> m_session_history_index { 0 };
-
-    // https://html.spec.whatwg.org/multipage/history.html#session-history
-    Vector<JS::NonnullGCPtr<SessionHistoryEntry>> m_session_history;
+    // https://html.spec.whatwg.org/multipage/document-sequences.html#browsing-context
+    JS::GCPtr<HTML::WindowProxy> m_window_proxy;
 
-    // https://html.spec.whatwg.org/multipage/browsers.html#creator-url
-    Optional<AK::URL> m_creator_url;
+    // https://html.spec.whatwg.org/multipage/browsers.html#opener-browsing-context
+    JS::GCPtr<BrowsingContext> m_opener_browsing_context;
 
-    // https://html.spec.whatwg.org/multipage/browsers.html#creator-base-url
-    Optional<AK::URL> m_creator_base_url;
+    // https://html.spec.whatwg.org/multipage/document-sequences.html#opener-origin-at-creation
+    Optional<HTML::Origin> m_opener_origin_at_creation;
 
-    // https://html.spec.whatwg.org/multipage/browsers.html#creator-origin
-    Optional<HTML::Origin> m_creator_origin;
+    // https://html.spec.whatwg.org/multipage/browsers.html#is-popup
+    TokenizedFeature::Popup m_is_popup { TokenizedFeature::Popup::No };
 
-    JS::GCPtr<HTML::NavigableContainer> m_container;
+    // https://html.spec.whatwg.org/multipage/document-sequences.html#is-auxiliary
+    bool m_is_auxiliary { false };
 
-    // https://w3c.github.io/webdriver/#dfn-window-handles
-    String m_window_handle;
+    // https://html.spec.whatwg.org/multipage/document-sequences.html#browsing-context-initial-url
+    Optional<AK::URL> m_initial_url;
 
-    // https://html.spec.whatwg.org/multipage/browsers.html#browsing-context
-    JS::GCPtr<HTML::WindowProxy> m_window_proxy;
+    // https://html.spec.whatwg.org/multipage/document-sequences.html#virtual-browsing-context-group-id
+    u64 m_virtual_browsing_context_group_id = { 0 };
 
+    // FIXME: Move cursor tracking to Navigable
     JS::GCPtr<DOM::Position> m_cursor_position;
     RefPtr<Core::Timer> m_cursor_blink_timer;
     bool m_cursor_blink_state { false };
```

Comment:
I didn't touch on it in this PR, but another candiate is 'has_been_discarded()`. It was added by @trflynn89  in 2d5381fd916f346222faa7ab8f4ec0f2cdef3e08 for WebDriver to use, and used to be called from the BrowsingContext::discard() function. However, we don't have that anymore. We'll likely need to move this state to Navigable (or TraversableNavigable) as well.